pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -259,13 +259,19 @@ func (r *AtlasDeploymentReconciler) handleDeploymentBackupSchedule(ctx *workflow
 	}
 	apiScheduleRes.Policies = []mongodbatlas.Policy{apiPolicy}
 
-	currentSchedule, _, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
+	currentSchedule, response, err := ctx.Client.CloudProviderSnapshotBackupPolicies.Delete(context.Background(), projectID, cName)
 	if err != nil {
-		r.Log.Debugf("unable to delete current backup policy for project: %v:%v, %v", projectID, cName, err)
+		errMessage := "unable to delete current backup policy for project"
+		r.Log.Debugf("%s: %v:%v, %v", errMessage, projectID, cName, err)
+		return fmt.Errorf("%s: %v:%v, %w", errMessage, projectID, cName, err)
 	}
 	if currentSchedule == nil {
 		r.Log.Debugf("got empty response for deleting backup schedule for project: %v:%v", projectID, cName)
-		return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api: %v", currentSchedule)
+		if response != nil {
+			return fmt.Errorf("can't delete сurrent backupschedule. got an empty response from the api. response status: %s", response.Status)
```

Comment:
`got an empty response from the api` seems like it should be removed from both here and from of the `else` statement. If the request was made `response` is never empty, if the response is empty, the request was never made, but the `error` was supposed to be returned (meaning we will `return` prior to this line).

So we can assume that response is not `nil` on this line, no need to check.

To be fair I'm not sure this logic is correct as `response` is returned both for success and failures, meaning

Reply:
Done